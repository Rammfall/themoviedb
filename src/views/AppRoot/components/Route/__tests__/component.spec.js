import React from 'react'
import { shallow } from 'enzyme'

import RouteComponent from '../component'

describe('RouteComponent matches snapshot', () => {
  const defaultProp = {
    exact: true,
    path: '/',
    // eslint-disable-next-line react/display-name
    component: () => <div>Test</div>,
    redirectTo: '/',
    conditionRender: false
  }

  it('with default props', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(<RouteComponent {...defaultProp} />)

    expect(component).toMatchSnapshot()
  })

  it('with conditionRender', () => {
    const props = {
      ...defaultProp,
      conditionRender: true
    }
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(<RouteComponent {...props} />)

    expect(component).toMatchSnapshot()
  })
})
