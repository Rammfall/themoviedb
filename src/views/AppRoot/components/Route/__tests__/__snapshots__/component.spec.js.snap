// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteComponent matches snapshot with conditionRender 1`] = `
<Route
  exact={true}
  path="/"
>
  <component />
</Route>
`;

exports[`RouteComponent matches snapshot with default props 1`] = `
<Route
  exact={true}
  path="/"
>
  <Redirect
    to="/"
  />
</Route>
`;
